packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -430,6 +455,40 @@ class _InsightsHandler {
         }
     }
 
+    getTestRunId(context: any): string|undefined {
+        if (!context) {return}
+
+        if (context.currentTest) {
+            const uniqueIdentifier = getUniqueIdentifier(context.currentTest, this._framework)
+            return this._tests[uniqueIdentifier] && this._tests[uniqueIdentifier].uuid
+        }
+
+        if (!context.test) {
+            return
+        }
+
+        const getTestRunIdFromSuite = (parent: any): string|undefined => {
+            if (!parent) {
+                return
+            }
+            for (const test of parent.tests) {
+                const uniqueIdentifier = getUniqueIdentifier(test, this._framework)
+                if (this._tests[uniqueIdentifier]) {
+                    return this._tests[uniqueIdentifier].uuid
+                }
+            }
+
+            for (const suite of parent.suites) {
+                const testRunId: string|undefined = getTestRunIdFromSuite(suite)
+                if (testRunId) {
+                    return testRunId
+                }
+            }
+            return
+        }
+        return getTestRunIdFromSuite(context.test.parent)
```

Comment:
same here
```suggestion
        if (!context.test.parent) {
            return
        }
        for (const test of context.test.parent.tests) {
            const uniqueIdentifier = getUniqueIdentifier(test, this._framework)
            if (this._tests[uniqueIdentifier]) {
                return this._tests[uniqueIdentifier].uuid
            }
        }

        for (const suite of context.test.parent.suites) {
            const testRunId: string|undefined = getTestRunIdFromSuite(suite)
            if (testRunId) {
                return testRunId
            }
        }
        return
```

Reply:
@christian-bromann it's a recursive function, that it is getting called from itself. It gets the test run id recursively, so that's why had to make it a function

Reply:
Have moved to a different function in the class, as that method requires the instance variables of the class. https://github.com/webdriverio/webdriverio/commit/5cfd96f8c5cb186ea62e97e0659b71e83c630b43